ViewLayer/DebugDisplay/MotorFaultPage/MotorFaultView/MotorFaultView.cpp

```diff
@@ -0,0 +1,2819 @@
+    dclReducedDueToLowPackVoltage_.setFixedSize(current_widthB, height);
+    layoutB->addWidget(&dclReducedDueToLowPackVoltage_);
+    dclReducedDueToLowPackVoltage_.hide();
+
+    dclReducedDueToLowSoc_.resize(current_widthB, height);
+    dclReducedDueToLowSoc_.setText("DCL Reduced Due To Low SOC");
+    dclReducedDueToLowSoc_.setStyleSheet(limitStyleSheet);
+    dclReducedDueToLowSoc_.setFixedSize(current_widthB, height);
+    layoutB->addWidget(&dclReducedDueToLowSoc_);
+    dclReducedDueToLowSoc_.hide();
+
+    dclReducedDueToTemperature_.resize(current_widthB, height);
+    dclReducedDueToTemperature_.setText("DCL Reduced Due To Temperature");
+    dclReducedDueToTemperature_.setStyleSheet(limitStyleSheet);
+    dclReducedDueToTemperature_.setFixedSize(current_widthB, height);
+    layoutB->addWidget(&dclReducedDueToTemperature_);
+    dclReducedDueToTemperature_.hide();
+
+    ui_.motor0ContentsWidget().setLayout(layout0);
+    ui_.motor1ContentsWidget().setLayout(layout1);
+    ui_.batteryContentsWidget().setLayout(layoutB);
+
+    connectMotorFaults(motorFaultsPresenter_);
+    connectBatteryFaults(batteryFaultsPresenter_);
+}
+
+MotorFaultView::~MotorFaultView()
+{
+}
+
+void MotorFaultView::connectMotorFaults(MotorFaultsPresenter& motorFaultsPresenter)
+{
+    connect(&motorFaultsPresenter, SIGNAL(motorZeroErrorFlagsReceived(ErrorFlags)),
+            this, SLOT(motorZeroErrorFlagsReceived(ErrorFlags)));
+    connect(&motorFaultsPresenter, SIGNAL(motorZeroLimitFlagsReceived(LimitFlags)),
+            this, SLOT(motorZeroLimitFlagsReceived(LimitFlags)));
+    connect(&motorFaultsPresenter, SIGNAL(motorOneErrorFlagsReceived(ErrorFlags)),
+            this, SLOT(motorOneErrorFlagsReceived(ErrorFlags)));
+    connect(&motorFaultsPresenter, SIGNAL(motorOneLimitFlagsReceived(LimitFlags)),
+            this, SLOT(motorOneLimitFlagsReceived(LimitFlags)));
+}
+
+void MotorFaultView::connectBatteryFaults(BatteryFaultsPresenter& batteryFaultsPresenter)
+{
+    connect(&batteryFaultsPresenter, SIGNAL(errorFlagsReceived(BatteryErrorFlags)),
+            this, SLOT(errorFlagsReceived(BatteryErrorFlags)));
+    connect(&batteryFaultsPresenter, SIGNAL(limitFlagsReceived(BatteryLimitFlags)),
+            this, SLOT(limitFlagsReceived(BatteryLimitFlags)));
+}
+
+void MotorFaultView::motorZeroErrorFlagsReceived(ErrorFlags motorZeroErrorFlags)
+{
+    static bool badMotorPositionHallSequenceOn = false;
+    static bool configReadErrorOn = false;
+    static bool dcBusOverVoltageOn = false;
+    static bool desaturationFaultOn = false;
+    static bool motorOverSpeedOn = false;
+    static bool railUnderVoltageLockOutOn = false;
+    static bool softwareOverCurrent = false;
+    static bool watchdogCausedLastReset = false;
+
+    int current_width = ui_.motor0ContentsWidget().size().width();
+    int current_height = ui_.motor0ContentsWidget().size().height();
+
+    if (motorZeroErrorFlags.badMotorPositionHallSequence())
```

Comment:
Let's encapsulate this if/else statement as a function as well.
`updateLabel(bool& RecievedValue, bool& CurrentValue, QLabel& label`

Reply:
Wouldn't let me use reference for receivedValue